src/map/Map.js

```diff
@@ -657,9 +657,23 @@ export var Map = Evented.extend({
 		var onResponse = Util.bind(this._handleGeolocationResponse, this),
 		    onError = Util.bind(this._handleGeolocationError, this);
 
-		if (options.watch) {
+		if (options.getCoordinates) {
+			if (options.watch) {
+				options.getCoordinates(options)
+					.then(function (locationWatchId) {
+						this._locationWatchId = locationWatchId;

```

Comment:
Are we sure 'this' Is referring to the context now it is wrapped in an anonymous function.  If the intent is to set _locationWatchId on the map object I think this needs to be an arrow function, or, bind the function to control the context of this?